dev/src/reference.ts

```diff
@@ -1458,6 +1458,33 @@ export class VectorQuerySnapshot<
   }
 }
 
+interface QueryStreamResponse<
+  AppModelType = firestore.DocumentData,
+  DbModelType extends firestore.DocumentData = firestore.DocumentData,
+> {
+  transaction?: Uint8Array;
+  readTime?: Timestamp;
+  explainMetrics?: ExplainMetrics;
+  document?: QueryDocumentSnapshot<AppModelType, DbModelType>;
+}
+
+interface QueryResponse<TSnapshot> {
+  transaction?: Uint8Array;
+  readTime?: Timestamp;
+  explainMetrics?: ExplainMetrics;
+  result: TSnapshot;
```

Comment:
The `result` snapshot is optional. If running `explain(..)` without execution, it should be null. However, it is also possible to run `explain(..)` with execution results.